### DISCUSSION TITLE
jaeger-ui release process

### DISCUSSION THREAD

**Discussion by sudheerv** [jaegertracing/jaeger#3238] [Q&A] [answered]
Hi - 

Can someone share the official release process for jaeger-ui project? We are evaluating forking the project internally and maintaining (potentially customizing etc) locally. Is there any official documented release process for the project that describes the frequency of major/minor/patch release cadence and how the updates affect compatibility (both backward and forward). Thanks!

**Comment by yurishkuro** [comment, marked as answer]
We do not treat jaeger-ui as a standalone product, its release cycle is tied to the main Jaeger backend release.  As such, there are no compatibility guarantees to either the data formats expected by the UI or the internal APIs (e.g. React components). We had discussions about exposing a more stable (and more restricted) internal API compatibility via an extra abstraction level, but it was never implemented.

The only compatibility guarantee that exist is when embedding the UI via iframe, done by products like Kiali, which is at the level of URLs.

**Comment by jkowall** [comment]
> Hi -
> 
> Can someone share the official release process for jaeger-ui project? We are evaluating forking the project internally and maintaining (potentially customizing etc) locally. Is there any official documented release process for the project that describes the frequency of major/minor/patch release cadence and how the updates affect compatibility (both backward and forward). Thanks!

What kind of changes are you trying to make? Maybe there is a way to contribute those upstream and make the project better for all, along with solving the specific requirements that make you want to fork it. Maintaining a fork is no small feat.
